uvm/bootstrap/com/untangle/uvm/UvmContextSelector.java

```diff
@@ -0,0 +1,248 @@
+     * @param context LoggerContext
+     */
+    @Override
+    public void removeContext(LoggerContext context) {
+        synchronized (contexts) {
+            LoggerContext remove = contexts.remove(context.getName());
+        }
+    }
+
+    /**
+     * Overrides getLoggerContexts method of ContextSelector interface
+     * @return List<LoggerContext>
+     */
+    @Override
+    public List<LoggerContext> getLoggerContexts() {
+        Collection<LoggerContext> values = contexts.values();
+        return new ArrayList<>(values);
+    }
+
+    /**
+     * Set the current thread's logging config to the "App" settings
+     * @param appId appId
+     */
+    public void setLoggingApp(Long appId) {
+        this.setThreadLoggingInformation("app-" + appId.toString());
+    }
+
+    /**
+     * Set the current thread's logging config to the "UVM" settings
+     */
+    public void setLoggingUvm() {
+        this.setThreadLoggingInformation("uvm");
+    }
+
+    /**
+     * Sets the current thread's logging context
+     * @param contextName contextName
+     */
+    private void setThreadLoggingInformation(String contextName) {
+        THREAD_LOG_INFO.set(contextName);
+    }
+
+    /**
+     * Causes all logging contexts to reconfigure themselves from
+     * the configuration context specified in the UvmLoggerContext.
+     */
+    public void reconfigureAll() {
+         synchronized (contexts) {
+             for (LoggerContext loggerContext : contexts.values()) {
+                 loggerContext.reconfigure();
+             }
+         }
+    }
+
+    /**
+     * A LoggerContext that associates the
+     * current UvmLoggerContext and allows configuration
+     * based on the contexts configuration file.
+     */
+    private static class UvmLoggerContext extends LoggerContext {
+        /**
+         * UvmLoggerContext constructor
+         * @param name name
+         */
+        public UvmLoggerContext(String name) {
+            super(name);
+        }
+
+        /**
+         * Overrides reconfigure method of LoggerContext
+         */
+        @Override
+        public void reconfigure() {
+            URI configLocation = null;
+            ConfigurationSource source = null;
+            try {
+                configLocation = getClass().getClassLoader().getResource("log4j.xml").toURI();
+                source = new ConfigurationSource(configLocation.toURL().openStream(), new File(configLocation));
+            } catch (URISyntaxException | IOException e) {
+                LOGGER.error("Exception: ", e);
+                throw new RuntimeException(e);
+            }
+            // Initialize the configuration
+            Configuration configuration = new XmlConfiguration(this, source);
+            configuration.initialize();
+            updateConfiguration(configuration, this.getName());
+            this.setConfiguration(configuration);
+        }
+
+        /**
+         * Updates the configuration according to contextName
```

Comment:
Elaborate more on what this method is doing

Reply:
Added additional information